packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -135,6 +135,62 @@ function initialize(obj, properties) {
 }
 
 /**
+  `CoreObject` is the base class for all Ember constructs. It establishes a
+  class system based on Ember's Mixin system, and provides the basis for the
+  Ember Object Model. `CoreObject` should generally not be used directly,
+  instead you should use `EmberObject`.
+
+  ## Usage
+
+  You can define a class by extending from `CoreObject` using the `extend`
+  method:
+
+  ```js
+  const Person = CoreObject.extend({
+    name: 'Tomster',
+  });
+  ```
+
+  For detailed usage, see the [Object Model](https://guides.emberjs.com/release/object-model/)
+  section of the guides.
+
+  ## Usage with Native Classes
+
+  Native JavaScript `class` syntax can be used to extend from any `CoreObject`
+  based class:
+
+  ```js
+  class Person extends CoreObject {
+    init() {
+      super.init(...arguments);
+      this.name = 'Tomster';
+    }
+  }
+  ```
+
+  Some notes about `class` usage:
+
+  * `new` syntax is not currently supported with classes that extend from
+    `EmberObject`. You must continue to use the `create` method when making new
```

Comment:
The example above refers to `CoreObject`, and the above sentence would seem to suggest that I can use `new Foo()` as long as I extend from `CoreObject` instead of `EmberObject`. Is this accurate?

Reply:
Good catch, that is not accurate. Will update to refer to `CoreObject` instead